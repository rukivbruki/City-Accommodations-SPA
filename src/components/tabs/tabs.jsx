import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {ActionCreator} from "../../reducer/data/data";
import TabsLink from "../tabs-link/tabs-link";
import {getCities} from "../../reducer/data/selector";

const Tabs = (props) => {
  const {
    currentCity,
    activeElement,
    changeCurrentCity,
    cities,
    onSelect,
  } = props;

  return (
    <div className="tabs">
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {cities.map((city, index) =>
            <TabsLink
              key={`${index}-${city}`}
              city={city}
              onSelect={onSelect}
              activeElement={activeElement || currentCity}
              changeCurrentCity={changeCurrentCity}
            />)}
        </ul>
      </section>
    </div>
  );
};

Tabs.propTypes = {
  activeElement: PropTypes.string,
  changeCurrentCity: PropTypes.func.isRequired,
  cities: PropTypes.arrayOf(PropTypes.string).isRequired,
  currentCity: PropTypes.string.isRequired,
  onSelect: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  cities: getCities(state)
});

const mapDispatchToProps = (dispatch) => ({
  changeCurrentCity: (city) => dispatch(ActionCreator.changeCity(city))
});

export {Tabs};
export default connect(mapStateToProps, mapDispatchToProps)(Tabs);
