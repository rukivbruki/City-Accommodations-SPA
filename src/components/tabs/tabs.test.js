import React from "react";
import renderer from "react-test-renderer";
import {Tabs} from "./tabs";
import Constants from "../../constants";

jest.mock(`../tabs-link/tabs-link.jsx`, () => jest.fn().mockReturnValue(null));

it(`Tabs component render correct`, () => {
  const tabsTemplate = renderer.create(
      <Tabs
        activeElement={`Paris`}
        changeCurrentCity={jest.fn()}
        cities={Constants.CITIES}
        currentCity={`Paris`}
        onSelect={jest.fn()}
      />
  ).toJSON();

  expect(tabsTemplate).toMatchSnapshot();
});
