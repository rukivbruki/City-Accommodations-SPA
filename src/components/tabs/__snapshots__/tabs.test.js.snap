// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component render correct 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    />
  </section>
</div>
`;
